pkg/util/azure/azure.go

```diff
@@ -35,6 +35,9 @@ func IsRunningOn() bool {
 
 // GetHostAlias returns the VM ID from the Azure Metadata api
 func GetHostAlias() (string, error) {
+	if !config.IsCloudProviderEnabled(CloudProviderName) {
```

Comment:
It could have a side effect on `diagnosis.DefaultCatalog`. I think it is probably OK but I did not check all the locations where `diagnosis.DefaultCatalog` is used.

Reply:
I'm unsure as well.  I don't see that diagnosis .DefaultCatalog makes an explicit call to the cloud provider endpoints, but I'm not extremely familiar with the codebase and can't say for certain.

@ogaca-dd is this worth further discussion in your opinion?